src/backports/three_way_comparison.h

```diff
@@ -0,0 +1,384 @@
+//
+// Created by kalmbacj on 9/2/25.
+//
+// Copyright 2025, Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
```

Comment:
Needs a proper header (see other files for examples, also feel free to add your name as a contributor here).